File: ABC/ABC104/c.py
from math import ceil
D, G = map(int, input().split(" "))
pc = [[int(i) for i in input().split(" ")] for _ in range(D)]

s = 100 * 11
for i in range(2 ** D):
    ans = 0
    num = 0
    zero = 0
    for j in range(D):
        if (i >> j) & 1:
            ans += 100*(j+1)*pc[j][0] + pc[j][1]
            num += pc[j][0]
        else:
            zero = j
    if G > ans:
        cand = ceil((G-ans)/(100 * (zero+1)))
        if cand > pc[zero][0]:
            continue
        else:
            num += cand
    if num < s:
        s = num
print(s)

File: ABC/ABC104/c.cpp
#include <iostream>
using namespace std;

int main() {
    int D;
    long long int G;
    cin >> D >> G;

    int p[D], c[D];
    for (int i=0; i<D; ++i) cin >> p[i] >> c[i];

    int ans = 9999999;
    for (int i=0; i<(1<<D); ++i) {
        long long int score=0;
        int num=0;
        int maxmid = -1;
        for (int j=0; j<D; ++j) {
            if (i >> j & 1) {
                score += 100 * (j+1) * p[j] + c[j];
                num += p[j];
            } else maxmid = j;
        }
        if (score < G) {
            if ((G - score) <= 100 * (maxmid+1) * (p[maxmid]-1)) {
                num += ((G - score) + (100 * (maxmid + 1)) - 1) / (100 * (maxmid + 1)); 
            } else num = 9999999;
        }
        if (num < ans) ans = num;
    }

    cout << ans << endl;

    return 0;
}